lms/djangoapps/certificates/generation_handler.py

```diff
@@ -170,14 +171,20 @@ def _can_generate_certificate_common(user, course_key, enrollment_mode):
         log.info(f'{user.id} : {course_key} does not have an enrollment. Certificate cannot be generated.')
         return False
 
-    if not modes_api.is_eligible_for_certificate(enrollment_mode):
+    is_eligible_for_cert = modes_api.is_eligible_for_certificate(enrollment_mode)
+    if not is_eligible_for_cert:
         log.info(f'{user.id} : {course_key} has an enrollment mode of {enrollment_mode}, which is not eligible for a '
                  f'certificate. Certificate cannot be generated.')
         return False
 
     if _required_verification_missing(course_key, user):
-        log.info(f'{user.id} does not have a verified id. Certificate cannot be generated for {course_key}.')
-        return False
+        # Honor courses can be eligible for certificates and don't require a valid approved IDV
+        if is_eligible_for_cert and enrollment_mode in CourseMode.NON_VERIFIED_MODES:
```

Comment:
is_eligible_for_cert is always True here because line 175-178 returns early if it's not. 

super nit:  Rather than 
```
if:
    if:
        # log
    else:
        return
```
I think it's easier to read:
```
if:
    if not:
       return
    # log
```
because it keeps the same "fail early or continue" flow as the rest of the function. I don't feel strongly on it though.

Reply:
Agreed! Thanks for the feedback. I changed it to fit the second form. Thanks, Matt!